### PR TITLE
keep-moving blog post added

### PR DESCRIPTION
**Pull request by jonnywildey**
keep on moving es5 es6 tips

### CODE REVIEW

File: _posts/2016-06-01-keep-on-moving.markdown [hunk 1]
```diff
@@ -0,0 +1,65 @@
+---
+layout: post
+title:  "Keep on Moving - ES5 -> ES6 Tips"
+date:   2016-06-20 10:00:00 +0100
+author: Jonny Wildey
+---
+
+A few months ago we decided to fully commit to the new **ES6**. Although the transition has been relatively smooth I thought I'd share some of the problems we found and some of the solutions we had to them.
```
Comment (jonnyarnold) [inline _posts/2016-06-01-keep-on-moving.markdown]: Could we link "ES6" to something like [this guide](https://github.com/lukehoban/es6features)?

File: _posts/2016-06-01-keep-on-moving.markdown [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+**ES6**
+
+```
+this.exaggerate = function(str) {
+  return str + '!'
+};
+const statements = ['cool', 'no worries', 'sure'];
+const exaggeratedSentences = _.map(statements, st => this.exaggerate(st) );
+
+```
```
Comment (jonnyarnold) [inline _posts/2016-06-01-keep-on-moving.markdown]: I think ```javascript adds syntax highlighting?

File: _posts/2016-06-01-keep-on-moving.markdown [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+## Put a lid on it
+
+By the time we decided to move to Node 6 we already had a substantial codebase, so migrating all of our code to the new norms was not going to be possible. We adopted the approach of that though we may not be able to change all of our code, we should at least be moving towards full implementation. How do we do this?
```
Comment (jonnyarnold) [inline _posts/2016-06-01-keep-on-moving.markdown]: Another possible link is "Node 6"?

Comment (jonnyarnold) [inline _posts/2016-06-01-keep-on-moving.markdown]: > We adopted the approach of that though we may not be able to change all of our code, we should at least be moving towards full implementation.

This sentence doesn't scan for me. How about...

> We adopted an approach that allowed us to move towards the full implementation without having to do it right now.

File: _posts/2016-06-01-keep-on-moving.markdown [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+To a certain extent we use linting, but in a repo with 20 files using ES6 and 160 not, this tends to flood your terminal with warnings rather than direct you to the problems that should be solved.
+
+We decided to create a set of tests called _lids_. These run through our source files, looking for ES6 files. All functions and files using ES6 syntax should have a
```
Comment (jonnyarnold) [inline _posts/2016-06-01-keep-on-moving.markdown]: Where does the name lids come from?

File: _posts/2016-06-01-keep-on-moving.markdown [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+```
+'use strict';
+```
```
Comment (jonnyarnold) [inline _posts/2016-06-01-keep-on-moving.markdown]: Maybe just inline this?

> All functions and files using ES6 syntax should have a `use strict` so this is pretty easy to implement.

File: _posts/2016-06-01-keep-on-moving.markdown [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+# Worthwhile?
+
+This was quick to implement, the majority of the work being on file navigation. It's been surprisingly effective, with all of the developers being caught out every now and then. And it turns of ES6 adoption, view this unnecessary graph!
+
+![Downward sloping graph](/images/keep-moving/useful-graph.png)
+
+
+## When do I refactor?
+
+We originally made a rule that if you touched a file, you were responsible for refactoring it. Unfortunately this tended to create some HUGE pull requests and be incredibly unfair for anyone who had a task that made minor changes to multiple files. Now we tend to refactor one or two files per pull request. It doesn't muddy up the original intent of the PR and is still progress.
```
Comment (jonnyarnold) [inline _posts/2016-06-01-keep-on-moving.markdown]: We should end with a summary of the article's main findings. We should probably point people to our Twitter feed if they want to comment.

**Comment by jonnyarnold** [comment]
Nice article!

File: _posts/2016-06-01-keep-on-moving.markdown (hunk 1 above)
Comment (jonnywildey) [inline _posts/2016-06-01-keep-on-moving.markdown]: 👍

File: _posts/2016-06-01-keep-on-moving.markdown (hunk 2 above)
Comment (jonnywildey) [inline _posts/2016-06-01-keep-on-moving.markdown]: Will do

File: _posts/2016-06-01-keep-on-moving.markdown (hunk 3 above)
Comment (jonnywildey) [inline _posts/2016-06-01-keep-on-moving.markdown]: 👌

File: _posts/2016-06-01-keep-on-moving.markdown (hunk 4 above)
Comment (jonnywildey) [inline _posts/2016-06-01-keep-on-moving.markdown]: It comes from Pascal, he uses "Put a lid on it" quite a lot. Was going to mention that but then I'd have to explain Pascal's relationship to us, and I was feeling tired...

File: _posts/2016-06-01-keep-on-moving.markdown (hunk 6 above)
Comment (jonnywildey) [inline _posts/2016-06-01-keep-on-moving.markdown]: Will do

**Comment by jonnyarnold** [comment]
Looks great! :+1:

**Pull request merged**